### PR TITLE
[Golang] Support multiple APIKeys

### PR DESCRIPTION
**Pull request by nmuesch**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

This PR aims to address #797 by adding multiple values to the Context attribute passed to authentication. This allows us to use a unique key to reference the Context with to retrieve any number of values for `ApiKey`

### CODE REVIEW

**Comment by wing328** [comment]
@nmuesch thanks for the PR. I wonder if you can update the Go (XML) petstore sample:
```
 // BasicAuth provides basic http authentication to a request passed via context using ContextBasicAuth
Perform git status
On branch pull/3210
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   samples/client/petstore/go/go-petstore-withXml/api_fake_classname_tags123.go
	modified:   samples/client/petstore/go/go-petstore-withXml/api_pet.go
	modified:   samples/client/petstore/go/go-petstore-withXml/api_store.go
	modified:   samples/client/petstore/go/go-petstore-withXml/configuration.go
```

Script: `./bin/go-petstore-withxml.sh`

File: modules/openapi-generator/src/main/resources/go/configuration.mustache [hunk 1]
```diff
@@ -26,7 +26,11 @@ var (
 	ContextAccessToken = contextKey("accesstoken")
 
 	// ContextAPIKey takes an APIKey as authentication for the request
-	ContextAPIKey = contextKey("apikey")
+	{{#authMethods}}
+    {{#isApiKey}}
+		Context{{name}} = contextKey("{{name}}")
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/go/configuration.mustache]: Seems like this line is not indented correctly (1 additional tab)

Comment (wing328) [reply modules/openapi-generator/src/main/resources/go/configuration.mustache]: And what about making ContextAPIKey a map instead as that's what we do in oher languages?

**Comment by wing328** [comment]
Also the current implementation will result in compilation errors if the security name contains space or dash (e.g. "api key", "api-key")

**Comment by nmuesch** [comment]
Thanks for the input and review! I plan on getting back to this shortly and will give it another iteration.

**Comment by wing328** [comment]
@nmuesch just ping me via https://gitter.im if you need any help.

**Comment by wing328** [comment]
Please submit the change (breaking) against `go-experimental` generator instead. 

Closing this one for the time being.

**Pull request closed without merging**